x-pack/plugins/security/server/authentication/authenticator.test.ts

```diff
@@ -57,31 +54,35 @@ describe('Authenticator', () => {
     mockBasicAuthenticationProvider = {
       login: jest.fn(),
       authenticate: jest.fn(),
-      logout: jest.fn(),
+      logout: jest.fn().mockResolvedValue(DeauthenticationResult.notHandled()),
       getHTTPAuthenticationScheme: jest.fn(),
     };
 
     jest.requireMock('./providers/http').HTTPAuthenticationProvider.mockImplementation(() => ({
+      type: 'http',
       authenticate: jest.fn().mockResolvedValue(AuthenticationResult.notHandled()),
+      logout: jest.fn().mockResolvedValue(DeauthenticationResult.notHandled()),
     }));
 
-    jest
-      .requireMock('./providers/basic')
-      .BasicAuthenticationProvider.mockImplementation(() => mockBasicAuthenticationProvider);
+    jest.requireMock('./providers/basic').BasicAuthenticationProvider.mockImplementation(() => ({
+      type: 'basic',
+      ...mockBasicAuthenticationProvider,
+    }));
+
+    jest.requireMock('./providers/saml').SAMLAuthenticationProvider.mockImplementation(() => ({
+      type: 'saml',
+      getHTTPAuthenticationScheme: jest.fn(),
+    }));
   });
 
   afterEach(() => jest.clearAllMocks());
 
   describe('initialization', () => {
     it('fails if authentication providers are not configured.', () => {
-      expect(() => new Authenticator(getMockOptions())).toThrowError(
-        'No authentication provider is configured. Verify `xpack.security.authc.providers` config value.'
-      );
-    });
-
-    it('fails if configured authentication provider is not known.', () => {
```

Comment:
__note:__ I'll return this test (it was hard to make with manual config mocking, but since I use real `createConfig` in tests it's easy).